src/tlo/methods/hiv.py

```diff
@@ -1117,25 +1117,27 @@ def update_parameters_for_program_scaleup(self):
             p["prob_prep_for_agyw"] = scaled_params["prob_prep_for_agyw"]
             p["probability_of_being_retained_on_prep_every_3_months"] = scaled_params["probability_of_being_retained_on_prep_every_3_months"]
 
+            # perfect retention on ART
+            p["probability_of_being_retained_on_art_every_3_months"] = scaled_params["probability_of_being_retained_on_art_every_3_months"]
+
             # increase probability of VMMC after hiv test
             p["prob_circ_after_hiv_test"] = scaled_params["prob_circ_after_hiv_test"]
 
             # increase testing/diagnosis rates, default 2020 0.03/0.25 -> 93% dx
-            p["hiv_testing_rates"]["annual_testing_rate_children"] = scaled_params["annual_testing_rate_children"]
             p["hiv_testing_rates"]["annual_testing_rate_adults"] = scaled_params["annual_testing_rate_adults"]
 
             # ANC testing - value for mothers and infants testing
             p["prob_hiv_test_at_anc_or_delivery"] = scaled_params["prob_hiv_test_at_anc_or_delivery"]
             p["prob_hiv_test_for_newborn_infant"] = scaled_params["prob_hiv_test_for_newborn_infant"]
 
-            # prob ART start if dx, this is already 95% at 2020
-            p["prob_start_art_after_hiv_test"] = scaled_params["prob_start_art_after_hiv_test"]
-
             # viral suppression rates
             # adults already at 95% by 2020
             # change all column values
             p["prob_start_art_or_vs"]["virally_suppressed_on_art"] = scaled_params["virally_suppressed_on_art"]
 
+            # update exising linear models to use new scaled-up paramters
+            self.pre_initialise_population()
```

Comment:
good logic. I think I'll factor it out into `self.build_lms()` (that is called from `self.pre_initialise_population` in case there are other things that need to go into `self.pre_initialise_population` that we don't want run at this point.

Reply:
I've refactored this